src/frequenz/channels/_select.py

```diff
@@ -3,9 +3,137 @@
 
 """Select the first among multiple Receivers.
 
-Expects Receiver class to raise `StopAsyncIteration`
-exception once no more messages are expected or the channel
-is closed in case of `Receiver` class.
+# Usage
+
+If you need to receiver different types of messages from different receivers, you need
+to know the source of a particular received value to know the type of the value.
+
+[`select()`][frequenz.channels.select] allows you to do that. It is an
+[async iterator][typing.AsyncIterator] that will iterate over the values of all
+receivers as they receive new values.
+
+It yields a [`Selected`][frequenz.channels.Selected] object that will tell you the
+source of the received message. To make sure the received value is *casted* to the
```

Comment:
Quoting from: https://www.wordreference.com/englishusage/cast
> The verb cast has several other meanings. Note that for all its meanings its past tense and past participle is cast, not ‘casted’.

```suggestion
source of the received message. To make sure the received value is *cast* to the

Reply:
looks like this one is pending.

Reply:
Oh, thanks for checking, it must have been lost in some rebase because I'm sure I updated the file at some point. Updated now.